lib/internal/modules/esm/resolve.js

```diff
@@ -373,8 +373,9 @@ function resolvePackageTargetString(
   }
 
   if (!StringPrototypeStartsWith(target, './')) {
-    if (internal && !StringPrototypeStartsWith(target, '../') &&
-        !StringPrototypeStartsWith(target, '/')) {
+    if (internal &&
+        target[0] !== '/' &&
+        !StringPrototypeStartsWith(target, '../')) {
```

Comment:
it might be faster still (altho much less clean) to do:
```suggestion
        (target[0] !== '.' || target[1] !== '.' || target[2] !== '/')) {
```

Reply:
I'd prefer keeping this PR to replacing startsWith/endsWith usage on chars and do a follow up these

Reply:
there is a chance that that change would be an over optimization. V8 may already be optimizing startsWith calls for multiple characters.

Reply:
and you could consider `target.slice(0,3) === '../'` too :)

Reply:
(although that would need the primordial)

Reply:
oh right lol